include/deal.II/fe/fe_rannacher_turek.h

```diff
@@ -0,0 +1,87 @@
+#ifndef dealii__fe_rannacher_turek_h
+#define dealii__fe_rannacher_turek_h
+
+#include <deal.II/base/polynomials_rannacher_turek.h>
+#include <deal.II/fe/fe_poly.h>
+#include <deal.II/fe/fe_base.h>
+#include <string>
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+/**
+ * Implementation of Rannacher-Turek elements. Functions generated by this
+ * element will be discontinuous, but their jump along faces is mean value
+ * free.
+ *
+ * Implemented only in dimension 2, lowest order, without hanging nodes and
+ * restriction/prolongation.
+ *
+ * <h3>Interpolation</h3>
+ *
+ * <h4>Node values</h4>
+ * The @ref GlossNodes "node values" are moments on faces.
+ *
+ * <h4>Generalized support points</h4>
+ * To calculate the node values, we are using a QGauss rule on each face.
+ * By default, we are using a two point rule to integrate Rannacher-Turek
+ * functions exactly. But in order to be able to interpolate other
+ * functions with sufficient accuracy, the number of quadrature points
+ * used on a face can be adjusted in the constructor.
+ *
+ * @ingroup fe
+ * @author Patrick Esser
+ * @date 2015
+ */
+template <int dim>
+class FE_RannacherTurek : public FE_Poly<PolynomialsRannacherTurek<dim>, dim>
+{
+public:
+  /**
+   * Constructor for Rannacher-Turek element of degree @p degree, using
+   * @p n_face_support_points quadrature points on each face for
+   * interpolation. Notice that the element of degree 0 contains
+   * polynomials of degree 2.
+   *
+   * Only implemented for degree 0 in 2D.
+   */
+  FE_RannacherTurek(const unsigned int degree = 0,
+                    const unsigned int n_face_support_points = 2);
+
+  virtual std::string get_name() const;
+  virtual FiniteElement<dim> *clone() const;
+
+  virtual void interpolate(
+    std::vector<double> &local_dofs,
+    const std::vector<double> &values) const;
```

Comment:
Ah, fair enough. Another rather poorly named function in one of our base classes :-(